homeassistant/components/serial/sensor.py

```diff
@@ -65,26 +67,49 @@ async def async_added_to_hass(self):
 
     async def serial_read(self, device, rate, **kwargs):
         """Read the data from the port."""
-        reader, _ = await serial_asyncio.open_serial_connection(
-            url=device, baudrate=rate, **kwargs
-        )
+        logged_error = False
         while True:
-            line = await reader.readline()
-            line = line.decode("utf-8").strip()
-
             try:
-                data = json.loads(line)
-                if isinstance(data, dict):
-                    self._attributes = data
-            except ValueError:
-                pass
-
-            if self._template is not None:
-                line = self._template.async_render_with_possible_json_value(line)
-
-            _LOGGER.debug("Received: %s", line)
-            self._state = line
-            self.async_write_ha_state()
+                reader, _ = await serial_asyncio.open_serial_connection(
+                    url=device, baudrate=rate, **kwargs
+                )
+            except SerialException as e:
+                self._state = None
+                self._attributes = None
+                self.async_write_ha_state()
+                if not logged_error:
+                    _LOGGER.exception("Unable to connect to the serial device %s: %s. Will retry", device, e)
+                    logged_error = True
+                await asyncio.sleep(5)
```

Comment:
We can break out this part into a helper method since it's almost the same actions as in the exception block below.

Reply:
Agreed. I'd rather you do it, so I won't break anything.

Reply:
Ok.